hazelcast/src/main/java/com/hazelcast/spi/impl/operationservice/impl/Invocation.java

```diff
@@ -438,6 +443,10 @@ boolean detectAndHandleBackupTimeout(long timeoutMillis) {
             return false;
         }
 
+        if (shouldFailOnOperationBackupTimeout()) {
+            complete(new OperationBackupAckTimeoutException());
```

Comment:
Apart from the return, can you try to add as much context as possible to the exception? The regular OperationTimeoutException has a very long message containing a ton of data to give a bit of a clue what happens when the exception is thrown.